api/src/main/java/org/pmiops/workbench/actionaudit/auditors/AdminAuditorImpl.java

```diff
@@ -134,4 +134,50 @@ public void fireUnlockWorkspaceAction(long workspaceId) {
 
     actionAuditService.send(event);
   }
+
+  public void firePublishWorkspaceAction(
+      long workspaceId, String updatedCategory, String prevCategoryIfAny) {
+    DbUser dbUser = userProvider.get();
+    String actionId = actionIdProvider.get();
+    long timestamp = clock.millis();
+
+    ActionAuditEvent event =
+        ActionAuditEvent.builder()
+            .actionId(actionId)
+            .actionType(ActionType.PUBLISH)
+            .agentType(AgentType.ADMINISTRATOR)
+            .agentEmailMaybe(dbUser.getUsername())
+            .agentIdMaybe(dbUser.getUserId())
+            .targetType(TargetType.WORKSPACE)
+            .targetIdMaybe(workspaceId)
+            .targetPropertyMaybe("Category")
+            .previousValueMaybe(prevCategoryIfAny)
+            .newValueMaybe(updatedCategory)
+            .timestamp(timestamp)
+            .build();
+
+    actionAuditService.send(event);
+  }
+
+  public void fireUnpublishWorkspaceAction(long workspaceId, String category) {
```

Comment:
`prevCategory` would make the behavior of this param a little clearer.  (AdminAuditor.java too)